packages/ckeditor5-engine/src/model/treewalker.ts

```diff
@@ -183,6 +183,28 @@ export default class TreeWalker implements Iterable<TreeWalkerValue> {
 		}
 	}
 
+	/**
+	 * Moves treewalker {@link #position} to provided `position`. Treewalker will
+	 * continue traversing from that position.
+	 *
+	 * If the provided position is before the start boundary, the position will be
+	 * set to the start boundary. If the provided position is after the end boundary,
+	 * the position will be set to the end boundary.
+	 * This is done to prevent the treewalker from traversing outside the boundaries.
+	 *
+	 * @param position Position to jump to.
+	 */
+	public jumpTo( position: Position ): void {
```

Comment:
I am worried that this method does not clone the position.

I know we should avoid cloning due to possible performance problems. However, it may lead to nasty bugs after using this function, when:

*   tree walker will keep on iterating and will change the position, that you may want to use at some later point,
*   or you will change that position offset and affect the tree walker.

So far, we don't have a guideline whether you (as integrator/developer) should always pass a "new" (cloned) position to API, or whether the functions will do it for you. We can add this note do API docs too, to help avoid creating unnecessary positions.

At the beginning of the performance initiative, I checked how how big impact has creating so many position clones on the overall performance. I was surprised how many positions we clone (a few hundreds thousands for some of the test samples). But I was also surprised that cutting it in half* didn't really have much impact. And in the profiler, I couldn't see that `Position` constructor takes a lot of time. Not sure now, though, as we optimized a lot of stuff, and the percentage gains may be more significant.

\* - this change turned to be bugged anyway, but still, I could measure the performance change.